src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java

```diff
@@ -2,19 +2,41 @@
 
 import de.unistuttgart.iste.gits.common.event.UserProgressLogEvent;
 import de.unistuttgart.iste.gits.generated.dto.Content;
+import de.unistuttgart.iste.gits.generated.dto.RewardChangeReason;
 import de.unistuttgart.iste.gits.reward.persistence.dao.AllRewardScoresEntity;
 import de.unistuttgart.iste.gits.reward.persistence.dao.RewardScoreEntity;
+import de.unistuttgart.iste.gits.reward.persistence.dao.RewardScoreLogEntry;
 import org.springframework.stereotype.Component;
 
-import java.util.List;
+import java.time.OffsetDateTime;
 
+import java.time.temporal.ChronoUnit;
+import java.util.Collections;
+import java.util.List;
 @Component
 public class FitnessScoreCalculator implements ScoreCalculator {
+    private static final double MAX_DECREASE_PER_DAY = 20;
+
     @Override
     public RewardScoreEntity recalculateScore(AllRewardScoresEntity allRewardScores, List<Content> contents) {
         RewardScoreEntity fitnessScoreBefore = allRewardScores.getFitness();
 
-        // TODO calculate fitness score
+        UserProgressLogEvent event = UserProgressLogEvent.builder().build();
```

Comment:
why are you creating an event? There should be no event. If you need an event then your logic is not right

Reply:
removed it